experimental/dds/tree/.eslintrc.js

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+module.exports = {
+	extends: ['@fluidframework/eslint-config-fluid/eslint7'],
+	// This setting is the same as in fluid. However, specifying it here causes the parser module to be evaluated relative
+	// to shared-tree, which references @typescript-eslint/parser@~4.2.0 rather than @2.17.0. The former supports more advanced array
+	// types which are used in several modules.
+	parser: '@typescript-eslint/parser',
```

Comment:
I believe Fluid is on the 4.2 parser, possibly can be removed?  (I'll follow up after initial commit.)

Reply:
There's something weird happening with the 'no-unused-vars' rule where there are a lot of false positives unless we override the parser in the local .eslintrc.js.

A theory: because 'package.json' does not explicitly include the eslint plugins as dev dependencies, so we're getting random packages from the hoisted /node_modules/.

Reply:
Did either of you come to any conclusions? I can look into this as well. It's possible I misdiagnosed what was going wrong, but the comment is accurate to what I remember--before specifying the parser explicitly, the type declarations for 

The comment is at least accurate to what I remember: `@fluidframework/eslint-config-fluid` has a dependency on `@typescript-eslint/parser@~2.17.0`, and without specifying this here in the Whiteboard-internal repository I saw issues around the parser understanding some newer typescript features (specifically tuple types).

I would be a bit surprised if removing this still caused `@typescript-eslint/parser@~4.2.0` to be used, given that information.

Reply:
My PR #4233 should upgrade the shared config. I just need a review so it should be merged soon.

Reply:
Disabling 'no-unused-vars' for now (to match the fluid config) passes.